usr/lib/pkcs11/ep11_stdll/ep11_specific.c

```diff
@@ -5160,6 +5162,7 @@ static CK_RV ep11_ende_crypt_init(STDLL_TokData_t * tokdata, SESSION * session,
         ctx->context = ep11_state;
         ctx->context_len = ep11_state_l;
         if (rc != CKR_OK) {
+            encr_mgr_cleanup(ctx);
```

Comment:
This change doesnt match the patch desciption. Accident ?

Reply:
in general i would prefer the length checks were done first and output buffer would be left unmodified in case its too small. however, since pkcs11 doesnt mandate this, im okay with it.

Reply:
The output buffer isn't changed (there is no output buffer), just the internal context. We could certainly setup the context only if rc==OK, and just free the ep11_state otherwise.
Do you want me to change this?

Reply:
the comment was meant as a closing remark regarding the ep11_get_mechanism_list discussion. Somehow put it in the wrong place.

Reply:
Ah I see. The ep11_get_mechanism_list would have to run through the loop 2 times to achieve this. Once to get the number of elements required, and a second time to fill the list. And theoretically the list of mechs reported by EP11 could have changed in-between...